### PR TITLE
Explorer: auto create folder when file path ends with /

### LINKED ISSUE #173902: Explorer: auto create folder when file path ends with `/`
For now on `explorer.newFile` vscode doesn't distinguish wether inputbox has trailing `/` or not. It would create file instead of folder in that case, which can be handy.

For now if you run `explorer.newFile` had an idea to create new file within new folder `test/integration/etc` and then forgot test name to create or decided to do it later you cannot just create folder from typed value. You have either:

- Create dummy file at end eg: `test/integration/etc/a` and then remove just created file
- Select all & copy typed value, run `explorer.newFolder` and paste value

I want clean and fast way to switch between creating file or folder. Please see my implementation for desired behavior: https://github.com/microsoft/vscode/pull/173901

### PR DESCRIPTION
**Pull request by zardoy**
Fixes https://github.com/microsoft/vscode/issues/173902

### CODE REVIEW

**Comment by lramos15** [comment]
I believe https://github.com/microsoft/vscode/pull/162893 contains this

**Comment by zardoy** [comment]
Totally makes sense, I just first try with a setting when it comes to user experience improved as I expect higher review/accept rate in this case. Anyway, now this PR has just 3 LOC 😄

**Review by lramos15 — APPROVED** [review]

**Review by Tyriar — APPROVED** [review]

File: src/vs/workbench/contrib/files/browser/files.contribution.ts [hunk 1]
```diff
@@ -478,6 +478,11 @@ configurationRegistry.registerConfiguration({
 			description: nls.localize('explorer.incrementalNaming', "Controls what naming strategy to use when a giving a new name to a duplicated Explorer item on paste."),
 			default: 'simple'
 		},
+		'explorer.autoCreateNewFolder': {
+			'type': 'boolean',
+			description: nls.localize('explorer.autoCreateNewFolder', "Wether to create folder when new file name input value ends with slash (`/`)."),
```
Comment (alirazacs) [inline src/vs/workbench/contrib/files/browser/files.contribution.ts]: Whether typo

Comment (zardoy) [reply src/vs/workbench/contrib/files/browser/files.contribution.ts]: Emm, okay I guess. The code was removed, how did you leave the review?

**Pull request merged**